src/Modules/OrchardCore.Commerce/Views/OrderPart_Edit.cshtml

```diff
@@ -1,63 +1,122 @@
 
-<section id="shopping-cart">
-    <div class="panel-heading">
-        <label>@T["Ordered Items"]</label>
-    </div>
+@inject IMoneyService MoneyService
+@inject ISession Session
+@inject IPredefinedValuesProductAttributeService AttributesService
+
+@{
+    var availableCurrencyCodes = MoneyService.Currencies.Select(currency => currency.CurrencyIsoCode);
+    var defaultCurrencyCode = MoneyService.DefaultCurrency.CurrencyIsoCode ?? "USD";
 
-    @if (Model.LineItems?.Any() == true)
+    // Get all predefined attributes in the system by filtering for Products that have PriceVariantsPart attached.
+    var allProducts = await Session.Query<ContentItem, ProductPartIndex>().ListAsync();
+    var allPriceVariantProducts = allProducts.Where(product => product.As<PriceVariantsPart>() != null);
+
+    var availableAttributes = new Dictionary<string, List<string>>();
+    foreach (var priceVariant in allPriceVariantProducts)
     {
-        <div class="container">
-            <div class="d-none d-sm-flex row border-bottom mt-3 pb-2 font-weight-bold">
-                <div class="col-xs-12 col-sm-2">@T["Quantity"]</div>
-                <div class="col-xs-12 col-sm-4">@T["Product"]</div>
-                <div class="col-xs-12 col-sm-4">@T["Attributes"]</div>
-                <div class="col-xs-12 col-sm-1">@T["Price"]</div>
-            </div>
+        var attributes = AttributesService.GetProductAttributesRestrictedToPredefinedValues(priceVariant);
+        foreach (var attr in attributes)
+        {
+            var predefinedStrings = new List<string>();
+
+            var predefinedValues = (attr.Settings as TextProductAttributeFieldSettings).PredefinedValues;
+            predefinedStrings.AddRange(predefinedValues.Select(value => value.ToString()));
+
+            availableAttributes.TryAdd(attr.Name, predefinedStrings);
 
-            @for (var line = 0; line < Model.LineItems.Count; line++)
-            {
-                var currentItem = Model.LineItems[line];
+            // Also need to add potential new values to an already existing key.
+            var existingValues = availableAttributes[attr.Name];
+            var valuesToAdd = predefinedStrings.Where(predefinedValue => !existingValues.Contains(predefinedValue));
+            availableAttributes[attr.Name].AddRange(valuesToAdd);
+        }
+    }
+
+    var serializedDictionary = JsonSerializer.Serialize(availableAttributes);
+}
 
+<section id="shopping-cart">
+    <div class="panel-heading">
+        <h3>@T["Ordered Items"]</h3>
+    </div>
+    <fieldset class="mb-3">
+        <label>@T["Product properties"]</label>
+        <ul>
+            <li class="hint">
+                @T["Currencies: Selected currencies need to be the same for all Products."]
+            </li>
+            <li class="hint">
+                @T["Attributes: Attributes are only to be selected for Price Variant Products. Only select a value for those attributes that are relevant to the product in question."]
```

Comment:
> Why are you able to select irrelevant attributes?

This was intended as a way to allow adding attributes to a Product that is being added as part of the current request, but it's irrelevant now as attributes are only populated for a Product after it's been added. Same for the mixed-together attribute values.

> It would be better if availableAttributes was Dictionary<string, Dictionary<string, List<string>>> so you get available attributes separately by SKU. Then you only list the applicable ones by lineItem.productSku.

Adjusted it to be this way.